src/Sylius/Bundle/ApiBundle/Serializer/CommandFieldItemIriToIdentifierDenormalizer.php

```diff
@@ -30,35 +31,61 @@ final class CommandFieldItemIriToIdentifierDenormalizer implements ContextAwareD
     /** @var CommandAwareInputDataTransformer */
     private $commandAwareInputDataTransformer;
 
-    /** @var CommandItemIriArgumentToIdentifierMapInterface */
-    private $commandItemIriArgumentToIdentifierMap;
-
     public function __construct(
         DenormalizerInterface $objectNormalizer,
         ItemIriToIdentifierConverterInterface $itemIriToIdentifierConverter,
-        CommandAwareInputDataTransformer $commandAwareInputDataTransformer,
-        CommandItemIriArgumentToIdentifierMapInterface $commandItemIriArgumentToIdentifierMap
+        CommandAwareInputDataTransformer $commandAwareInputDataTransformer
     ) {
         $this->objectNormalizer = $objectNormalizer;
         $this->itemIriToIdentifierConverter = $itemIriToIdentifierConverter;
         $this->commandAwareInputDataTransformer = $commandAwareInputDataTransformer;
-        $this->commandItemIriArgumentToIdentifierMap = $commandItemIriArgumentToIdentifierMap;
     }
 
     public function supportsDenormalization($data, $type, $format = null, array $context = [])
     {
-        return $this->commandItemIriArgumentToIdentifierMap->has($this->getInputClassName($context));
+        /** @psalm-var class-string $inputClassName|null */
+        $inputClassName = $this->getInputClassName($context);
+
+        if ($inputClassName === null) {
+            return false;
+        }
+
+        foreach (class_implements($inputClassName) as $classInterface) {
+            if ($classInterface === CommandFieldItemIriToIdentifierAwareInterface::class) {
+                return true;
+            }
+        }
+
+        return false;
     }
 
     public function denormalize($data, $type, $format = null, array $context = [])
     {
         /** @psalm-var class-string $inputClassName */
         $inputClassName = $this->getInputClassName($context);
 
-        $fieldName = $this->commandItemIriArgumentToIdentifierMap->get($inputClassName);
-
-        if (array_key_exists($fieldName, $data)) {
-            $data[$fieldName] = $this->itemIriToIdentifierConverter->getIdentifier($data[$fieldName]);
+        foreach (class_implements($inputClassName) as $classInterface) {
+            if ($classInterface === CommandFieldItemIriToIdentifierAwareInterface::class) {
+                foreach (get_class_vars($inputClassName) as $classFieldName => $classFieldValue) {
```

Comment:
I would be for iterating on data from the request and checking if there is an iri field, and that field should be converted to identifier